lib/actions-oclif/env/rm.ts

```diff
@@ -89,23 +89,23 @@ export default class EnvRmCmd extends Command {
 			);
 		}
 
-		return patterns
-			.confirm(
+		try {
+			await patterns.confirm(
 				options.yes || false,
 				'Are you sure you want to delete the environment variable?',
-			)
-			.then(function() {
-				if (options.device) {
-					return balena.pine.delete({
-						resource: 'device_environment_variable',
-						id: params.id,
-					});
-				} else {
-					return balena.pine.delete({
-						resource: 'application_environment_variable',
-						id: params.id,
-					});
-				}
-			});
```

Comment:
Block converted from promise `.then` to `async/await`, plus some code deduplication and improved error handling.